pallets/staking/src/lib.rs

```diff
@@ -47,17 +48,19 @@ impl<T: Config> SortedListProvider<T::AccountId> for EmptyList<T> {
 	type Score = u64;
 
 	fn iter() -> Box<dyn Iterator<Item = T::AccountId>> {
+		defensive!();
```

Comment:
The [docs](https://docs.rs/frame-support/14.0.0/frame_support/macro.defensive.html) for this macro say  

> Generic function to mark an execution path as ONLY defensive.
> Similar to mark a match arm or if/else branch as unreachable!.

- What does "**ONLY defensive**" mean here ?
- Is this supposed to be the same as `unreachable!` ? If yes what's the difference between the two and why not use `unreachable!` instead ?

Reply:
defensive calls will panic on unoptimized builds (builds on top of assert_debug). The idea is to have an unreachable behavior when testing while still printing logs on --release, we don't want nodes (unless testing where crashing is useful) to suddenly collapse if a bug went unnoticed. The idea is to make sure none of the unreachable paths are used so far. We will restrict the staking extrinsics to only make available the ones planned and tested with a call filter. Once we start fleshing out more functionality we may need to use some of these functions. Once the MVP is done we can refactor it.